src/mscorlib/shared/System/Lazy.cs

```diff
@@ -383,7 +383,7 @@ private void PublicationOnlyViaFactory(LazyHelper initializer)
         private void PublicationOnlyWaitForOtherThreadToPublish()
         {
             var spinWait = new SpinWait();
-            while (!ReferenceEquals(_state, null))
+            while (!(_state is null))
```

Comment:
The other places that check `_state` are still using ReferenceEquals. Using `is` here while keeping the rest on the ReferenceEquals plan does not look like an improvement.

Reply:
For comparing an object to a non-null object, is there any benefit of using ReferenceEquals over `==`? If not, I can change the other calls as well.

Reply:
`==` may use overriden operator Equals if there is one.

`ReferenceEquals` is explicit about your intent: that you want to compare references, and not use any overridden operators.